### PR TITLE
Create a custom button workshop

### PR DESCRIPTION
**Pull request by brianegan**
This PR introduces a workshop that teaches folks how to build a custom button with Flutter. 

Unlike the previous workshop PR, the content for this workshop has not been reviewed yet. Would be happy to get technical or content feedback on any step :)

## Pre-launch Checklist

- [X] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [X] I signed the [CLA].
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-devrel channel on [Discord].

[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by brianegan** [comment]
Hm, this is an interesting one! The CI is failing on this error:

```
error - src/custom_button/step_01/snippet.dart:10:18 - The function 'CustomButton' isn't defined. Try importing the library that defines 'CustomButton', correcting the name to the name of an existing function, or defining a function named 'CustomButton'. - undefined_function
```

However, that's kinda the point of the exercise haha... I want the user to see the error `'CustomButton' isn't defined`. Then, they should fix the error themselves. I actually ran into the same problem in the previous workshop as well -- sometimes the snippets are intentionally incomplete so folks can fill in the solution.

What do you think we should do in these cases?

**Comment by domesticmouse** [comment]
> What do you think we should do in these cases?

I'm not sure what the global answer is. One option is to ignore `undefined_function` for this file, the other is to put in a skeleton

**Comment by brianegan** [comment]
> I'm not sure what the global answer is. One option is to ignore undefined_function for this file, the other is to put in a skeleton.

Sounds good. In that case, I'll just try to do what I think is best for the end-user while still making the CI happy (important for long-term maintenance).

Thanks again for the help!

File: dartpad_codelabs/src/custom_button/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    has_solution: true
+  - name: Tidy Up
+    directory: step_08
+    has_solution: true
+  - name: Accessibility
+    directory: step_09
+    has_solution: true
+  - name: Review
+    directory: step_10
+    has_solution: false
```
Comment (domesticmouse) [inline dartpad_codelabs/src/custom_button/meta.yaml:33]: nit: please add new line at eof

**Review by domesticmouse — APPROVED** [review]

File: dartpad_codelabs/src/custom_button/meta.yaml (hunk 1 above)
Comment (brianegan) [reply dartpad_codelabs/src/custom_button/meta.yaml:33]: :+1:  Haha, trying out VSCode instead of IntelliJ... forgot this setting :) Fixed!

**Comment by domesticmouse** [comment]
@brianegan Are you ready for me to land this PR?

**Comment by brianegan** [comment]
@domesticmouse I wasn't sure if there was any feedback on the content itself, but if it looked good then it's ready to merge from my side!

**Comment by domesticmouse** [comment]
Thanks @brianegan!

**Pull request merged**